src/data_capture/video_capture/scripts/capture.py

```diff
@@ -48,15 +118,22 @@ def _is_record_callback(self, data):
         is_record = data.data
         try:
             if is_record:
-                rospy.loginfo("Starting to record video")
+                rospy.loginfo("Starting to record video")             
+
+                # Starting the periodic check on free memory, stops if exceeds 90% utilization at interval=2 seconds
+                # RepeatedTimer auto-starts, no need of rt.start()
+                rt = RepeatedTimer(2, monitorAvailableMemory, 90, self._video_recorder) 
```

Comment:
I think that all this should be outside of the video recorder. You can still tell it what to do on using too much memory, e.g., send a ROS message to the video recorder to shut down, etc.

Having this in the class, complicates the classes logic and there isn't necessarily a reason for these classes to be coupled, other than that the video is likely to use a lot of the RAM. It may be possible that in the future other tasks are eating RAM and shutting down the video recording isn't sufficient, for example, when Emily adds the audio processing. A more generic shutdown call would probably be better.

Also, if you have the monitor available memory class in its own node, you can avoid threading (much easier to test and less error prone). Also, you can use something like [`schedule`](https://pypi.org/project/schedule/) and avoid writing your own timer (just schedule to call a callback to check if you've exceeded RAM usage and if you have send a ROS message).